papertty/papertty.py

```diff
@@ -574,9 +574,13 @@ def list_drivers():
 @click.pass_obj
 def scrub(settings, size):
     """Slowly fill with black, then white"""
-    if size not in range(8, 32 + 1):
-        PaperTTY.error("Invalid stripe size, must be 8-32")
     ptty = settings.get_init_tty()
+    # If partial refresh isn't supported, don't try to stripe the scrub.
+    # Set the size to be the full width instead
+    if not ptty.driver.supports_partial:
+        size = ptty.driver.width
```

Comment:
This same logic is now in three different places, so putting it closer to scrub itself would be better: https://github.com/joukos/PaperTTY/blob/master/papertty/drivers/drivers_base.py#L60

The whole scrub feature is questionable though, originally I put it there because the only 2.13" I had at the time was actually faulty, it turned out (or, had its own personality, as these displays seem to have), and it was the only way for me to reliably get it back to its senses / clean again after some messing about :)

It shouldn't be needed with a non-partial refresh supporting display or even a partial one that can be cleared normally, but I guess it doesn't hurt to have something to "massage" the pixels every now and then.

Reply:
PR has been updated as suggested to perform the logic in one place, closer to the driver. It's been implemented as an overridden method in drivers_full.

I agree that the utility of scrubbing itself is questionable though. I've been using it while testing out different methods of performing partial draws which left the screen in a weird state, but outside of testing/debugging/developing I'm not sure what purpose it serves.